utility/utils.py

```diff
@@ -966,3 +966,39 @@ def get_random_str(size=13):
     """
     chars = string.ascii_lowercase + string.digits
     return ''.join(random.choice(chars) for _ in range(size))
+
+
+def run_async(command):
+    """
+    Run command locally and return without waiting for completion
+
+    Args:
+        command (str): The command to run.
+
+    Returns:
+        An open descriptor to be used by the calling function.
+
+    Example:
+        command = 'oc delete pvc pvc1'
+        proc = run_async(command)
+        ret, out, err = proc.async_communicate()
+    """
+    log.info(f"Executing command: {command}")
+    popen_obj = subprocess.Popen(
+        command, stdin=subprocess.PIPE, stdout=subprocess.PIPE, shell=True,
+        encoding='utf-8'
+    )
+
+    def async_communicate():
```

Comment:
dont think it is necessary to have this function as it is one liner at test code.

Reply:
Yes, open_obj can be used inside the test case to get the stdout and stderr using popen_obj.communicate() and then get return code using popen_obj.return code.
The method async_communicate just club those and make it simpler to use in testcase.
ret, out, err = popen_obj.async_communicate()

Reply:
missed the other function is a closure, thanks for clarifying.